dotcom-rendering/src/components/ElectionTrackers/ChangeBars.tsx

```diff
@@ -0,0 +1,239 @@
+import {
+	from,
+	headlineBold17Object,
+	headlineBold24Object,
+	headlineBold42Object,
+	headlineMedium17Object,
+	headlineMedium20Object,
+} from '@guardian/source/foundations';
+import { palette } from '../../palette';
+
+type Props = {
+	changes: Change[];
+};
+
+type Change = {
+	/**
+	 * The name of the value changing. For an election, this could be the name
+	 * of the group or party. It will be used as a React "key" for the element,
+	 * so each value's `name` should be unique relative to the names of other
+	 * changes.
+	 *
+	 * **Examples:** name of a candidate; name of a party.
+	 */
+	name: string;
+	/**
+	 * An abbreviated version of the {@linkcode Change.name}, used when the full
+	 * name will not fit, for example on narrower breakpoints. Note that due to
+	 * the constraints of the design this cannot be longer than about 7
+	 * characters.
+	 */
+	abbreviation: string;
+	/**
+	 * The change to be represented. Can be positive, negative or zero.
+	 */
+	change: number;
+	/**
+	 * The colour used to represent the value. It expects a CSS `color` value
+	 * (e.g. a hex string). To ensure dark mode support a {@linkcode palette}
+	 * colour can be used; i.e. this property can be set to the return value of
+	 * the {@linkcode palette} function.
+	 */
+	colour: string;
+};
+
+export const ChangeBars = ({ changes }: Props) => (
+	<ul
+		css={{
+			display: 'flex',
+			margin: 0,
+			padding: '4px 0',
+			height: 170,
+		}}
+	>
+		{changes.map((change) => (
+			<ChangeBar
+				key={change.name}
+				change={change}
+				maxChange={Math.max(...changes.map((c) => c.change))}
```

Comment:
Could you compute `maxChange` once as derived state in the main render path of `ChangeBars`?

Reply:
Good spot 👍